ngapp/src/app/student/story-drawer/story-drawer.component.ts

```diff
@@ -121,4 +131,96 @@ export class StoryDrawerComponent implements OnInit {
       }
     });
   }
+
+  /**
+   * Open a dialog asking the user if they really want to delte their story
+   * Call the function to delete the story if the user clicks 'yes'
+   * @param id story id to be deleted
+   */
+  openDeleteStoryDialog(id: string) {
+    this.dialogRef = this.dialog.open(BasicDialogComponent, {
+      data: {
+        title: this.ts.l.delete_story,
+        message: "Are you sure you want to delete this story?",
+        confirmText: this.ts.l.yes,
+        cancelText: this.ts.l.no,
+      },
+      width: "50vh",
+    });
+
+    this.dialogRef.afterClosed().subscribe(async (res) => {
+      this.dialogRef = undefined;
+      if (res) {
+        this.deleteStory(id);
+      }
+    });
+  }
+
+  /**
+   * Delete the given story and any associated recordings
+   * Set the new 'current story' => next one in the list, or first one
+   * if the last story in the list was deleted
+   * @param id story id to be deleted
+   */
+  deleteStory(id: string) {
+    // remove any associated recordings
+    this.recordingService.deleteStoryRecordingAudio(id).subscribe((_) => {});
+    this.recordingService.deleteStoryRecording(id).subscribe((_) => {});
+
+    // get index of story to be deleted within story list
+    let storyIndex = this.stories.findIndex((story) => story._id === id);
```

Comment:
Nit: I think this can be `const`

```suggestion
    const storyIndex = this.stories.findIndex((story) => story._id === id);
```